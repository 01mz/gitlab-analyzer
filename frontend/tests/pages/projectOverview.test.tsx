import React from 'react';
import Index from '../../pages/project/[projectId]/overview/index';
import {mount, ReactWrapper} from "enzyme";

describe("Overview", () =>{
    const useRouter = jest.spyOn(require('next/router'), 'useRouter');
    const mockUseEffect = jest.spyOn(React, 'useEffect')
    const mockAxios = jest.spyOn(require('axios'), 'get');
    const mockEnqueue = jest.spyOn(require('notistack'), "useSnackbar");
    let enqueueSnackbar = jest.fn();
    let rend:ReactWrapper

    beforeEach(async()=>{
        mockEnqueue.mockImplementation(() => {return {enqueueSnackbar}});
        useRouter.mockImplementationOnce(() => ({
            query: { projectId: 'TestId' }, isReady: true
        }));
        rend = mount(
            <Index />
        );
        await Promise.resolve();
    })

    it("Snapshot projectId", () => {
        expect(rend).toMatchSnapshot();
    })

    it("Test useEffect", ()=>{
        expect(mockUseEffect).toBeCalled();
    })

    it('Test axios',()=>{
        expect(mockAxios).toBeCalled();
    })

})