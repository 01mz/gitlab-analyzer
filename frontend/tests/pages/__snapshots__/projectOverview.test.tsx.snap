// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Snapshot projectId 1`] = `
<index>
  <AuthView>
    <LoadingPage>
      <Styled(MuiBox)
        alignItems="center"
        display="flex"
        height="100vh"
        justifyContent="center"
        width="100vw"
      >
        <div
          className="MuiBox-root MuiBox-root-1"
        >
          <WithStyles(ForwardRef(CircularProgress))
            color="primary"
            size="7em"
          >
            <ForwardRef(CircularProgress)
              classes={
                Object {
                  "circle": "MuiCircularProgress-circle",
                  "circleDeterminate": "MuiCircularProgress-circleDeterminate",
                  "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
                  "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
                  "circleStatic": "MuiCircularProgress-circleStatic",
                  "colorPrimary": "MuiCircularProgress-colorPrimary",
                  "colorSecondary": "MuiCircularProgress-colorSecondary",
                  "determinate": "MuiCircularProgress-determinate",
                  "indeterminate": "MuiCircularProgress-indeterminate",
                  "root": "MuiCircularProgress-root",
                  "static": "MuiCircularProgress-static",
                  "svg": "MuiCircularProgress-svg",
                }
              }
              color="primary"
              size="7em"
            >
              <div
                className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                role="progressbar"
                style={
                  Object {
                    "height": "7em",
                    "width": "7em",
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </ForwardRef(CircularProgress)>
          </WithStyles(ForwardRef(CircularProgress))>
        </div>
      </Styled(MuiBox)>
    </LoadingPage>
  </AuthView>
</index>
`;
